### PR TITLE
image: add 127.0.0.1.nip.io to DefaultAllowedRegistriesForImport

### PR DESCRIPTION
**Pull request by mfojtik**
fixes: https://github.com/openshift/origin/issues/15033

### CODE REVIEW

**Comment by mfojtik** [comment]
@soltysh quick review?

**Comment by smarterclayton** [comment]
This is a security issue, potentially.

**Comment by mfojtik** [comment]
@smarterclayton only if somebody manage to run registry on 127.0.0.1?

**Comment by mfojtik** [comment]
@openshift/sig-security PTAL

File: pkg/cmd/server/api/types.go [hunk 1]
```diff
@@ -154,6 +154,7 @@ var (
 		{DomainName: "quay.io"},
 		// FIXME: Probably need to have more fine-tuned pattern defined
 		{DomainName: "*.amazonaws.com"},
+		{DomainName: "127.0.0.1.nip.io"},
```
Comment (enj) [inline pkg/cmd/server/api/types.go]: `*.127.0.0.1.nip.io`?

**Comment by enj** [comment]
This seems like it would be a safe thing to do...

File: pkg/cmd/server/api/types.go (hunk 1 above)
Comment (liggitt) [reply pkg/cmd/server/api/types.go]: does this let people target the localhost on production servers? this feels like it should be part of devenv setup

**Comment by smarterclayton** [comment]
Why don't you set the defaults based on your other PR?  Because this isn't going to work for oc cluster up (IP will be different), or openshift-ansible (not going to use it).

I don't think we can blanket trust the local cluster.  I think if we want to allow this it needs to be easy for an admin to change this value (i.e. config in master or something).

**Comment by mfojtik** [comment]
@smarterclayton you mean automatically add the external registry into the list? i think that is reasonable.

**Comment by smarterclayton** [comment]
That's not quite Christophe's scenario, but we try to avoid having non
production defaults.

On Aug 29, 2017, at 3:42 AM, Michal Fojtik <notifications@github.com> wrote:

@smarterclayton <https://github.com/smarterclayton> you mean automatically
add the external registry into the list? i think that is reasonable.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/15919#issuecomment-325581757>,
or mute
the thread

File: pkg/cmd/server/api/types.go (hunk 1 above)
Comment (soltysh) [reply pkg/cmd/server/api/types.go]: I don't see any value in this, either.

**Pull request closed without merging**